src/models/Airline.ts

```diff
@@ -0,0 +1,357 @@
+                    }
+                }
+                this.aircraft[i].meta.liveTrack = false;
+            } else {
+                //check time
+                if (!aircraft.onGround
+                    && aircraft.barometricAltitude !== null
+                    && aircraft.barometricAltitude < PlaneAlert.config.thresholds.takeoff
+                    && this.aircraft[i].meta.onGround) {
+                    //Plane takeoff
+                    const nearestAirport = this.findNearestAirport(this.aircraft[i]);
+                    if (nearestAirport !== null) {
+                        PlaneAlert.log.info(`Plane ${this.name} (${this.aircraft[i].icao}) took off at ${nearestAirport.airport.name} (${nearestAirport.airport.gps_code})`);
+                    } else {
+                        PlaneAlert.log.info(`Plane ${this.name} (${this.aircraft[i].icao}) took off`);
+                    }
+                    this.triggerEvent(PlaneEvents.PLANE_TAKEOFF, {nearestAirport: nearestAirport?.airport}, this.aircraft[i]);
+                }
+                if (aircraft.onGround
+                    && aircraft.barometricAltitude !== null
+                    && aircraft.barometricAltitude < PlaneAlert.config.thresholds.landing
+                    && !this.aircraft[i].meta.onGround) {
+                    PlaneAlert.log.info(`Plane ${this.aircraft[i].icao} is landing`);
+                    //Plane landing
+                    const nearestAirport = this.findNearestAirport(this.aircraft[i]);
+                    if (nearestAirport !== null) {
+                        PlaneAlert.log.info(`Plane ${this.name} (${this.aircraft[i].icao}) landed at ${nearestAirport.airport.name} (${nearestAirport.airport.icao})`);
+                    } else {
+                        PlaneAlert.log.info(`Plane ${this.name} (${this.aircraft[i].icao}) landed`);
+                    }
+                    this.triggerEvent(PlaneEvents.PLANE_LAND, {nearestAirport: nearestAirport?.airport}, this.aircraft[i]);
+                }
+                PlaneAlert.log.info(`Plane ${this.name} (${this.aircraft[i].icao}) is ${aircraft.onGround ? "on ground" : "in the air"} at ${aircraft.latitude}, ${aircraft.longitude} with altitude ${aircraft.barometricAltitude}`);
+
+                this.aircraft[i].meta.liveTrack = true;
+                this.aircraft[i].meta.lastSeen = new Date().getTime();
+                this.aircraft[i].meta.onGround = aircraft.onGround;
+                this.aircraft[i].meta.lat = aircraft.latitude;
+                this.aircraft[i].meta.lon = aircraft.longitude;
+                this.aircraft[i].meta.alt = aircraft.barometricAltitude;
+                this.aircraft[i].meta.squawk = aircraft.squawk;
+            }
+
+            this.save();
+            // PlaneAlert.log.debug(`Plane ${this.name} (${this.icao}) returned data: ${JSON.stringify(data)}`);
+        }
+    }
+
+    private findNearestAirport(aircraft: AircraftMeta) {
```

Comment:
Since there is a new class now (not just Aircraft), this probably should be offloaded to an external helper class

Reply:
Common class maybe?